### DISCUSSION TITLE
"Illegal instruction" error with python import on Ubuntu

### DISCUSSION THREAD

**Discussion by asprasan** [mitsuba-renderer/mitsuba3#619] [Q&A] [answered]
I am using a PC with:
Ubuntu 20.04 
Intel(R) Xeon(R) W3565 CPU
x86_64 architecture

I installed mitsuba with pip inside a conda environment (python 3.10):
```sh
Collecting mitsuba
  Using cached mitsuba-3.2.1-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (33.5 MB)
Collecting drjit==0.4.1
  Using cached drjit-0.4.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.5 MB)
Installing collected packages: drjit, mitsuba
Successfully installed drjit-0.4.1 mitsuba-3.2.1
```

When I do `import mitsuba` inside python I get the error `Illegal Instruction` and it crashes.

I found others facing similar issue with Mac M1. But could not find any solutions for an Intel CPU.
What could be the possible issue behind this error?

Note: I have compiled mitsuba2 for another project of mine on the same PC

**Comment by njroussel** [comment, marked as answer]
The Intel CPU wheel you're installing through `pip` is targeting at least `ivybridge`. I guess you're missing some AVX instructions.
You will need to compile the project yourself if you want to use it on that CPU.

**Reply by asprasan** [reply]
Thank you very much. Building from source worked out for me.
